sql/policy_internal.sql

```diff
@@ -113,7 +115,37 @@ BEGIN
           _message = MESSAGE_TEXT,
           _detail = PG_EXCEPTION_DETAIL,
           _sqlstate = RETURNED_SQLSTATE;
-      RAISE WARNING 'converting chunk "%" to columnstore failed when columnstore policy is executed', chunk_rec.oid::regclass::text
+      RAISE WARNING 'converting chunk "%" to columnstore failed when recompress columnstore policy is executed', chunk_rec.oid::regclass::text
+          USING DETAIL = format('Message: (%s), Detail: (%s).', _message, _detail),
+                ERRCODE = _sqlstate;
+      chunks_failure := chunks_failure + 1;
+    END;
+    COMMIT;
+    BEGIN
+      -- went through recompression successfully now reindex indexes
+      IF (chunk_rec.status & bit_compressed_partial = bit_compressed_partial) AND (reindex_enabled IS TRUE) THEN
+        FOR idx_rec IN
+          SELECT schemaname, indexname
+          FROM pg_indexes
+          WHERE schemaname = chunk_rec.schema_name
+            AND tablename = chunk_rec.table_name
+            AND EXISTS (
+              SELECT 1
```

Comment:
```suggestion
              SELECT
```
Nit. Don't need to project anything here.